Editorial changes to MTF Examples (#41)
mulesoft/docs-mule-sdk: modules/ROOT/pages/mtf-testing-sources.adoc

include::_attributes.adoc[]
endif::[]

Apart from defining operations, you might want to define sources for your module.

> Since sources must be placed at the beginning of a flow, they can't appear directly in the test. When testing a source, you must have a flow with the source you wish to test, and then a test that triggers it.

Reviewer: ```suggestion
Because sources must be placed at the beginning of a flow, they can't appear directly in the test. To test a source, you must have a flow with the source you want to test, and then a test that triggers it.
```

> There are several types of sources, which also makes the testing strategies different for each of them.

Reviewer: ```suggestion
There are several types of sources, each with its own distinct testing strategy.
```

== Sources That Emit a Response

> The most common kind of test is when the source generates a synchronous response. To test this, you must hit the

Reviewer: I don't understand the first sentence. How does test=source generates a synchronous response? Are we testing to validate a synchronous response? to ensure a synchronous response? because we have received a synchronous response?

I'm afraid that I don't understand what it means to "hit the source" either, or to "assert" the response....

source to get that response, and then assert it.

> For example, in the case of an `http:listener` you can hit the source with an `http:request`:

Reviewer: An `http:listener` what? operation? 

ditto re. "hit the source"....

</munit:test>
----

== Polling Sources

> Since the polling source polls items periodically with a given `scheduling strategy`, the test must trigger the

Reviewer: ```suggestion
Because the polling source polls items periodically with a given scheduling strategy, the test must trigger the
```

> polling source and wait for the next poll. Then, it can assert the response generated by the source.

Reviewer: assert=validate?
assert=accept?

=== Trigger a Single Time

> For example, for a `file:listener` you can trigger the source by creating new files:

Reviewer: Because there is a subtopic title between this text and the previous, we should be more clear about "For example,..."  Example of what?

Reviewer: We should not use syntax or a name as a common noun: a `file:listener`
We must identify a `file:listener` what? As above, is this an operation?

</munit:test>
----

> === Trigger multiple times

Reviewer: title-style caps.
Also, we should not have two consecutive titles or headings without text between them. Insert some transition between the subtopic title and the first example.
